PLAN-2443 - Include allReduce in the Pattern docs where QPPs are described (#1407)
neo4j/docs-cypher: modules/ROOT/pages/functions/predicate.adoc

|===
| *Syntax* 3+| `allReduce(accumulator = initial, stepVariable IN list \| reductionFunction, predicate)`
| *Description* 3+| Returns true if, during the stepwise evaluation of a value across the elements in a given `LIST<ANY>`, the accumulated result satisfies a specified predicate at every step.

> Where that list is a xref:patterns/variable-length-patterns.adoc#group-variables[group variable] defined in a xref:patterns/variable-length-patterns.adoc#quantified-path-patterns[quantified path pattern] (not within a xref:patterns/shortest-paths.adoc[shortest path pattern]), it allows for the early pruning of paths that do not satisfy the predicate.

Reviewer: I would make it a bit longer and as a follow-up sentence, not between prentheses.

|===
| *Syntax* 3+| `allReduce(accumulator = initial, stepVariable IN list \| reductionFunction, predicate)`
| *Description* 3+| Returns true if, during the stepwise evaluation of a value across the elements in a given `LIST<ANY>`, the accumulated result satisfies a specified predicate at every step.

> When that list is a xref:patterns/variable-length-patterns.adoc#group-variables[group variable] defined in a xref:patterns/variable-length-patterns.adoc#quantified-path-patterns[quantified path pattern], its predicate will be inlined where applicable.

Reviewer 2: ```suggestion
If that list is a xref:patterns/variable-length-patterns.adoc#group-variables[group variable] defined in a xref:patterns/variable-length-patterns.adoc#quantified-path-patterns[quantified path pattern], its predicate is inlined where applicable.
```

> *Note*: AllReduce should not be used for a group variable within a xref:patterns/shortest-paths.adoc[shortest path pattern].

Reviewer: Sorry for being so picky but this doesn't sound quite right, for a few reasons:
* It should refer to the function as `allReduce()`, not AllReduce, see Considerations below,
* We try our very best to inline `allReduce()` predicates, but it isn't always possible,
* You can totally use it with `SHORTEST` but the important point is that it won't be inlined.

Something along the lines of:

When `allReduce()` is used with a xref:patterns/variable-length-patterns.adoc#group-variables[group variable] defined in a xref:patterns/variable-length-patterns.adoc#quantified-path-patterns[quantified path pattern], its predicate will be inlined where applicable. This inlining allows for early pruning of the search space by discarding paths as soon as the predicate is not satisfied. Note that `allReduce()` predicates are not inlined when used in a xref:patterns/shortest-paths.adoc[shortest path pattern], and therefore do not benefit from this pruning.
